### PR TITLE
f-checkout@0.51.0 - Handle PATCH response, and mock in Storybook

### PR DESCRIPTION
**Pull request by xander-marjoram**
### Added
- Axios mock to Storybook for checkout PATCH requests.
- PATCH response handling, by adding `issues` and `isFulfillable` to state.

---

## UI Review Checks

- [x] README and/or UI Documentation has been [created|updated]
- [x] Unit tests have been [created|updated]
- [x] This code has been checked with regard to [our accessibility standards](http://fozzie.just-eat.com/documentation/general/accessibility/checklist)

## Browsers Tested

- [x] Chrome (latest)
- [ ] Internet Explorer 11
- [ ] Mobile (Please list device/browser – Ideally one iPhone model and one Android)

### List any other browsers that this PR has been tested in:

- [View the Browser Support Checklist](http://fozzie.just-eat.com/documentation/general/browser-support)

### CODE REVIEW

**Review by kevinrodrigues — DISMISSED** [review]

**Review by zlatinivanov-je — DISMISSED** [review]

File: packages/components/organisms/f-checkout/src/store/tests/checkout.module.test.js [hunk 1]
```diff
@@ -183,6 +194,26 @@ describe('CheckoutModule', () => {
             });
         });
 
+        describe(`${UPDATE_IS_FULFILLABLE} ::`, () => {
```
Comment (dandel10n) [inline packages/components/organisms/f-checkout/src/store/tests/checkout.module.test.js]: Could all the tests from line 167 and until 226 be changes with one like this?: 
```
describe ('Mutation', () => {
    it.each([
        ['UPDATE_FULFILMENT_ADDRESS', address]
        ['UPDATE_IS_FULFILLABLE', isFulfillable],
        ['UPDATE_ISSUES', issues],
        etc
    ])(' "%s" should update state with received value'), (
    mutationName, value) => {
    }
})
```

Comment (xander-marjoram) [reply packages/components/organisms/f-checkout/src/store/tests/checkout.module.test.js]: @dandel10n It is possible, but you would need to add an extra (string) variable for referencing the state which I think is a little untidy.

```javascript
it.each([
    [UPDATE_FULFILMENT_ADDRESS, 'address', address],
    [UPDATE_FULFILMENT_TIME, 'time', time],
    [UPDATE_IS_FULFILLABLE, 'isFulfillable', isFulfillable],
    [UPDATE_ISSUES, 'issues', issues],
    [UPDATE_USER_NOTE, 'userNote', userNote]
])('%s :: should update state with received value', (mutationName, propertyName, propertyValue) => {
    // Arrange & Act
    mutations[mutationName](state, propertyValue);

    // Assert
    expect(state[propertyName]).toEqual(propertyValue);
});
```

Comment (dandel10n) [reply packages/components/organisms/f-checkout/src/store/tests/checkout.module.test.js]: I don't think that it looks untidy. In my opinion, this way looks much more maintainable and readable for me than adding a new test for every mutation. We can check all the mutations we have with one test and will need to add only one new line later for new mutation.

File: packages/components/organisms/f-checkout/src/demo/checkoutMock.js [hunk 2]
```diff
@@ -30,6 +33,9 @@ export default {
             case '/get-basket-collection.json':
                 mock.onGet(path).reply(200, getBasketCollection);
                 break;
+            case '/update-checkout.json':
+                mock.onPatch(updateCheckoutRegex).reply(200, updateCheckout);
```
Comment (kevinrodrigues) [inline packages/components/organisms/f-checkout/src/demo/checkoutMock.js:37]: Out of interest why does this need a path modifier. Doesn't the path get picked up from the mocked file passed in?

**Review by kevinrodrigues — APPROVED** [review]

**Review by dandel10n — APPROVED** [review]

**Pull request merged**